#! /bin/sh /usr/share/dpatch/dpatch-run
# Description: Build the X.Org driver only for the selected system X Server version.
# Author: Michael Meskes <meskes@debian.org>, Felix Geyer <debfx-pkg@fobos.de>

@DPATCH@
diff -urNad virtualbox-ose-3.1.0-dfsg~/src/VBox/Additions/common/crOpenGL/Makefile.kmk virtualbox-ose-3.1.0-dfsg/src/VBox/Additions/common/crOpenGL/Makefile.kmk
--- virtualbox-ose-3.1.0-dfsg~/src/VBox/Additions/common/crOpenGL/Makefile.kmk	2009-10-07 11:27:13.000000000 +0200
+++ virtualbox-ose-3.1.0-dfsg/src/VBox/Additions/common/crOpenGL/Makefile.kmk	2009-10-07 13:44:19.145463290 +0200
@@ -67,16 +67,11 @@
 VBoxOGL_INCS           = .
 if1of ($(KBUILD_TARGET),linux solaris)
  VBoxOGL_INCS     += \
-	$(VBOX_PATH_X11_ROOT)/libXdamage-1.1 \
-	$(VBOX_PATH_X11_ROOT)/libXcomposite-0.4.0 \
-	$(VBOX_PATH_X11_ROOT)/libXfixes-4.0.3 \
-	$(VBOX_PATH_X11_ROOT)/damageproto-1.1.0 \
-	$(VBOX_PATH_X11_ROOT)/compositeproto-0.4 \
-	$(VBOX_PATH_X11_ROOT)/fixesproto-4.0 \
-	$(VBOX_PATH_X11_ROOT)/libx11-1.1.5-other \
-	$(VBOX_PATH_X11_ROOT)/1.3/xorg \
+	/usr/include/x11 \
+	/usr/include/xorg \
+	/usr/include/pixman-1 \
 	$(VBOX_MESA_INCS) \
-	$(PATH_ROOT)/src/VBox/Additions/x11/x11include/libdrm-2.4.13
+	/usr/include/drm
  VBoxOGL_DEFS     += VBOX_NO_NATIVEGL
 endif
 
diff -Nur virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/Makefile.kmk virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/Makefile.kmk
--- virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/Makefile.kmk	2010-02-13 15:19:45.000000000 +0100
+++ virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/Makefile.kmk	2010-03-01 00:13:12.324136362 +0100
@@ -21,6 +21,10 @@
 SUB_DEPTH = ../../../..
 include $(KBUILD_PATH)/subheader.kmk
 
+ifn1of ($(XSERVER_VERSION), 70 71 13 14 15 16 17)
+ XSERVER_VERSION := 17
+endif
+
 # Include sub-makefiles.
 if1of ($(KBUILD_TARGET), freebsd linux netbsd openbsd solaris)
  include $(PATH_SUB_CURRENT)/VBoxClient/Makefile.kmk
diff -Nur virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxmouse/Makefile.kmk virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxmouse/Makefile.kmk
--- virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxmouse/Makefile.kmk	2010-02-13 15:20:20.000000000 +0100
+++ virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxmouse/Makefile.kmk	2010-03-01 00:05:56.464134113 +0100
@@ -27,7 +27,6 @@
 # vboxmouse_drv
 #
 if1of ($(KBUILD_TARGET), linux)
- SYSMODS += vboxmouse_drv
  vboxmouse_drv_TEMPLATE = VBOXGUESTR3XF86MOD
  vboxmouse_drv_DEFS.linux = linux
  vboxmouse_drv_DEFS.x86 += __i386__
@@ -63,7 +62,6 @@
 #
 # vboxmouse_drv_70
 #
-DLLS += vboxmouse_drv_70
 vboxmouse_drv_70_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_70_DEFS = \
 	XFree86Server IN_MODULE XFree86Module XFree86LOADER XINPUT XORG_7X
@@ -84,7 +82,6 @@
 #
 # vboxmouse_drv_71
 #
-DLLS += vboxmouse_drv_71
 vboxmouse_drv_71_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_71_DEFS := $(vboxmouse_drv_70_DEFS)
 vboxmouse_drv_71_INCS := \
@@ -101,7 +98,6 @@
 #
 # vboxmouse_drv_13
 #
-DLLS += vboxmouse_drv_13
 vboxmouse_drv_13_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_13_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_13_INCS := \
@@ -116,7 +112,6 @@
 #
 # vboxmouse_drv_14
 #
-DLLS += vboxmouse_drv_14
 vboxmouse_drv_14_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_14_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_14_INCS := \
@@ -138,7 +133,6 @@
 #
 # vboxmouse_drv_15
 #
-DLLS += vboxmouse_drv_15
 vboxmouse_drv_15_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_15_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_15_INCS := \
@@ -153,7 +147,6 @@
 #
 # vboxmouse_drv_16
 #
-DLLS += vboxmouse_drv_16
 vboxmouse_drv_16_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_16_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 vboxmouse_drv_16_INCS := \
@@ -169,7 +162,6 @@
 #
 # vboxmouse_drv_17
 #
-DLLS += vboxmouse_drv_17
 vboxmouse_drv_17_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxmouse_drv_17_DEFS := $(vboxmouse_drv_70_DEFS) NO_ANSIC
 ## @todo replace $(VBOX_PATH_X11_ROOT)/xorg-server-1.6.0-local
@@ -186,6 +178,14 @@
 	vboxmouse_15.c
 
 
+vboxmouse_drv_$(XSERVER_VERSION)_NAME := vboxmouse_drv
+vboxmouse_drv_$(XSERVER_VERSION)_INCS := \
+	/usr/include/x11 \
+	/usr/include/xorg \
+	/usr/include/pixman-1
+DLLS += vboxmouse_drv_$(XSERVER_VERSION)
+
+
 # Check the undefined symbols in the X.Org modules against lists of allowed
 # symbols.  Not very elegant, but it will catch problems early.
 ifdef VBOX_WITH_TESTCASES
diff -Nur virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxvideo/Makefile.kmk virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxvideo/Makefile.kmk
--- virtualbox-ose-3.1.4-dfsg~/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2010-02-13 15:20:20.000000000 +0100
+++ virtualbox-ose-3.1.4-dfsg/src/VBox/Additions/x11/vboxvideo/Makefile.kmk	2010-03-01 00:07:13.334156070 +0100
@@ -26,7 +26,6 @@
 # vboxvideo_drv
 #
 if1of ($(KBUILD_TARGET), linux)
- SYSMODS += vboxvideo_drv
  vboxvideo_drv_TEMPLATE = VBOXGUESTR3XF86MOD
  vboxvideo_drv_DEFS.linux = linux
  vboxvideo_drv_DEFS.x86 = __i386__
@@ -116,7 +115,6 @@
 #         base keywords instead of using .solaris or .linux.
 #         Also it is *important* to use := and not = when deriving a properity.
 #
-DLLS += vboxvideo_drv_70
 vboxvideo_drv_70_TEMPLATE = VBOXGUESTR3XORGMOD
 if1of ($(KBUILD_TARGET), linux)
  vboxvideo_drv_70_CFLAGS += \
@@ -139,7 +137,6 @@
 #
 # vboxvideo_drv_71
 #
-DLLS += vboxvideo_drv_71
 vboxvideo_drv_71_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_71_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_71_DEFS := $(vboxvideo_drv_70_DEFS)
@@ -155,7 +152,6 @@
 #
 # vboxvideo_drv_13
 #
-DLLS += vboxvideo_drv_13
 vboxvideo_drv_13_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_13_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_13_DEFS := $(vboxvideo_drv_70_DEFS)
@@ -177,7 +173,6 @@
 # This uses the same code (vboxvideo_13.c) as 1.3, but is built
 # for 1.4 as well in case there should be any relevant header changes.
 #
-DLLS += vboxvideo_drv_14
 vboxvideo_drv_14_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_14_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_14_DEFS := $(vboxvideo_drv_70_DEFS)
@@ -196,7 +191,6 @@
 #
 # vboxvideo_drv_15
 #
-DLLS += vboxvideo_drv_15
 vboxvideo_drv_15_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_15_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_15_DEFS := $(vboxvideo_drv_70_DEFS) NO_ANSIC PCIACCESS \
@@ -223,7 +217,6 @@
 #
 # vboxvideo_drv_16
 #
-DLLS += vboxvideo_drv_16
 vboxvideo_drv_16_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_16_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_16_DEFS := $(vboxvideo_drv_15_DEFS)
@@ -241,7 +234,6 @@
 #
 # vboxvideo_drv_17
 #
-DLLS += vboxvideo_drv_17
 vboxvideo_drv_17_TEMPLATE = VBOXGUESTR3XORGMOD
 vboxvideo_drv_17_CFLAGS := $(vboxvideo_drv_70_CFLAGS)
 vboxvideo_drv_17_DEFS := $(vboxvideo_drv_15_DEFS)
@@ -263,6 +255,16 @@
 vboxvideo_drv_17_SOURCES := $(vboxvideo_drv_15_SOURCES)
 
 
+vboxvideo_drv_$(XSERVER_VERSION)_NAME := vboxvideo_drv
+vboxvideo_drv_$(XSERVER_VERSION)_INCS := \
+	/usr/include/xorg \
+	/usr/include/x11 \
+	/usr/include/pixman-1 \
+	/usr/include/X11/dri \
+	/usr/include/drm
+DLLS += vboxvideo_drv_$(XSERVER_VERSION)
+
+
 # Check the undefined symbols in the X.Org modules against lists of allowed
 # symbols.  Not very elegant, but it will catch problems early.
 ifdef VBOX_WITH_TESTCASES
